src/uploadStrategy/strategy.ts

```diff
@@ -0,0 +1,18 @@
+import type {AxiosResponse} from 'axios';
+import type {BatchUpdateDocuments} from '../interfaces';
+
+export interface UploadStrategy {
+  /**
+   * Upload a batch of documents
+   *
+   * @param {BatchUpdateDocuments} batch
+   */
+  upload: (batch: BatchUpdateDocuments) => Promise<AxiosResponse>;
+
+  /**
+   * Async operation to run once the batch upload is complete
+   *
+   * @memberof UploadStrategy
+   */
+  postUpload?: () => Promise<void>;
```

Comment:
That's what I did initially. @louis-bompart thinks otherwise :P.
Personally, I don't care since the Strategy Interface is not exposed externally and I don't mind the optional chaining operator.

Reply:
We ☕ talked a bit with @olamothe.

The thing here is that either ones has to absorb the complexity, either the callee or the caller.

If the interface was exposed externally, the callee should absorb the complexity in the context of an SDK: Indeed, the SDK goal is to make things simpler to use, thus, the caller should have it easy.

In this case, because the interface is not exposed externally, it then goes I think to personal preferences.